cpp/src/lists/interleave_columns.cu

```diff
@@ -156,20 +193,25 @@ struct compute_string_sizes_and_interleave_lists_fn {
   }
 };
 
-/**
- * @brief Struct used in type_dispatcher to interleave list entries of the input lists columns and
- * output the results into a destination column.
- */
-struct interleave_list_entries_fn {
-  template <class T>
-  std::enable_if_t<std::is_same_v<T, cudf::string_view>, std::unique_ptr<column>> operator()(
-    table_view const& input,
-    column_view const& output_list_offsets,
-    size_type num_output_lists,
-    size_type num_output_entries,
-    bool data_has_null_mask,
-    rmm::cuda_stream_view stream,
-    rmm::mr::device_memory_resource* mr) const noexcept
+// Error case when no other overload or specialization is available
+template <typename T, typename Enable = void>
```

Comment:
This is interesting. What was the reasoning behind switching from function specializations to structs?

Reply:
This base case (which calls `CUDF_FAIL`) needs to cover all non-supported types. In the old implementation, any time we support a new type we also need to update this base case. The new implementation, on the other hand, can avoid that burden---we just need to implement support for the new type.